privacyidea/lib/tokens/hotptoken.py

```diff
@@ -832,7 +833,10 @@ def enroll_via_validate(cls, g, content, user_obj):
         # Create a challenge!
         c = token_obj.create_challenge()
         # get details of token
-        init_details = token_obj.get_init_detail(user=user_obj)
+        enroll_params = get_init_tokenlabel_parameters(g, user_object=user_obj,
+                                                       token_type=cls.get_class_type())
+        init_details = token_obj.get_init_detail(params=enroll_params,
+                                                 user=user_obj)
```

Comment:
since we don't go through [`create_challenges_from_tokens()`](https://github.com/privacyidea/privacyidea/blob/630066ef53e4bf8916a3c341dd91fe1768f5713c/privacyidea/lib/token.py#L2116) we are missing the `messages` attribute here.

Reply:
Done in fb0bb613

Reply:
in 11dc6504 it is even a list!